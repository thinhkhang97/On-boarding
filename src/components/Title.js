/// Trouble with font family.

import * as React from 'react';
import { Text } from 'react-native';

type Props = {
  style?: object
};

class Title extends React.Component<Props> {
  static defaultProps = {
    style: { fontSize: 32, fontWeight: 'bold', color: 'black' }
  };
  render() {
    return <Text style={this.props.style}>{this.props.children}</Text>;
  }
}

export default Title;
