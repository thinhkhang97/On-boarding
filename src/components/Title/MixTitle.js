// @flow

import * as React from 'react';
import { View, Text } from 'react-native';

type Props = {
  subTitleStyle?: View.propTypes.style,
  titleStyle?: View.propTypes.style,
  number: string,
  title: string,
  subTitle: string,
};

class MixTitle extends React.Component<Props> {
  static defaultProps = {
    subTitleStyle: {},
    titleStyle: {},
  };
  render() {
    const { title, subTitle, subTitleStyle, titleStyle } = this.props;
    return (
      <View
        style={{
          display: 'flex',
          alignItems: 'center',
          marginLeft: 15,
          marginRight: 15,
        }}
      >
        <Text
          style={[
            {
              fontSize: 32,
              fontWeight: 'bold',
              color: 'black',
            },
            titleStyle,
          ]}
        >
          {title}
        </Text>
        <Text style={[{ fontSize: 20, color: 'black' }, subTitleStyle]}>{subTitle}</Text>
      </View>
    );
  }
}

export default MixTitle;
