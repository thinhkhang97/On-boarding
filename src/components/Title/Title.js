// @flow
import * as React from 'react';
import { View, Text } from 'react-native';
import type { ____ViewStyleProp_Internal as Style } from 'react-native/Libraries/StyleSheet/StyleSheetTypes';

type Props = {
  style?: Style,
  children?: React.Node,
};

class Title extends React.Component<Props> {
  static defaultProps = {
    style: {},
  };
  render() {
    const { style, children } = this.props;
    return (
      <Text style={[{ fontSize: 32, fontWeight: 'bold', color: 'black' }, style]}>{children}</Text>
    );
  }
}

export default Title;
