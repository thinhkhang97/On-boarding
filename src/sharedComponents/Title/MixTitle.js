/// Trouble with font family.

import * as React from 'react';
import { View, Text } from 'react-native';

type Props = {
  numberStyle?: object,
  titleStyle?: object,
  number: string
};

class MixTitle extends React.Component<Props> {
  static defaultProps = {
    numberStyle: {},
    titleStyle: {}
  };
  render() {
    const { title, subTitle, numberStyle, titleStyle } = this.props;
    return (
      <View
        style={{
          display: 'flex',
          alignItems: 'center',
          marginLeft: 15,
          marginRight: 15
        }}
      >
        <Text
          style={[
            {
              fontSize: 32,
              fontWeight: 'bold',
              color: 'black'
            },
            numberStyle
          ]}
        >
          {title}
        </Text>
        <Text style={[{ fontSize: 20, color: 'black' }, titleStyle]}>
          {subTitle}
        </Text>
      </View>
    );
  }
}

export default MixTitle;
