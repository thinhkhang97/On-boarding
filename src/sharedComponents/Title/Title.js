/// Trouble with font family.
// @flow
import * as React from 'react';
import { View, Text } from 'react-native';

type Props = {
  style?: View.propTypes.style,
  children?: React.Node,
};

class Title extends React.Component<Props> {
  static defaultProps = {
    style: {},
  };
  render() {
    const { style, children } = this.props;
    return (
      <Text style={[{ fontSize: 32, fontWeight: 'bold', color: 'black' }, style]}>{children}</Text>
    );
  }
}

export default Title;
