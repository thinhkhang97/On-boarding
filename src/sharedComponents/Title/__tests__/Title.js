import * as React from 'react';
import renderer from 'react-test-renderer';
import Title from '../Title';

test('Render title correctly', ()=>{
    const testTitle = renderer.create(<Title/>).toJSON();
    expect(testTitle).toMatchSnapshot();
})