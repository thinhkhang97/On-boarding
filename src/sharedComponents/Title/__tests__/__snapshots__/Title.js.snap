// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render title correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "black",
        "fontSize": 32,
        "fontWeight": "bold",
      },
      Object {},
    ]
  }
/>
`;
